src/clj/puppetlabs/trapperkeeper/services/protocols/filesystem_watch_service.clj

```diff
@@ -4,8 +4,11 @@
 
 (def Event
   "Schema for an event on a file watched by this service."
-  {:type (schema/enum :create :modify :delete :unknown)
-   :path File})
+  (schema/if #(= (:type %) :unknown)
+    {:type (schema/eq :unknown)
+     :path (schema/eq nil)}
```

Comment:
Cool, I didn't find a good example of ensuring `nil` (lots of `schema/maybe`) with a quick skim. Happy to change that. Could you link me to an example?